pkg/rehearse/jobs.go

```diff
@@ -203,25 +209,114 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 				continue
 			}
 
-			exists, index, templateKey := hasChangedTemplateVolume(rehearsal.Spec.Containers[0].VolumeMounts, rehearsal.Spec.Volumes, templates)
+			exists, index, cmName, err := getVolumeIndexAndTempCMName(rehearsal.Spec.Containers[0].VolumeMounts, rehearsal.Spec.Volumes, templates)
+			if err != nil {
+				loggers.Job.WithError(err).Warn(skipJobMessage)
+				continue
+			}
+
 			if exists {
-				if templateData, err := config.GetTemplateData(templates[templateKey]); err != nil {
-					jobLogger.WithError(err).WithField("template-name", templates[templateKey].Name).Warn("couldn't get template's data. Job won't be rehearsed")
-					continue
-				} else {
-					templateName := config.GetTemplateName(templateKey)
-					rehearsal.Spec.Volumes[index].VolumeSource.ConfigMap.Name = config.GetTempCMName(templateName, templateKey, templateData)
-				}
+				rehearsal.Spec.Volumes[index].VolumeSource.ConfigMap.Name = cmName
 			}
 
 			jobLogger.WithField(logRehearsalJob, rehearsal.Name).Info("Created a rehearsal job to be submitted")
 			rehearsals = append(rehearsals, rehearsal)
 		}
 	}
 
+	// for the remaining cluster types of each template, pick a `random` corresponding job to rehearse.
+	for templateFile, clusterTypes := range templatesWithClusterTypes {
```

Comment:
I really think this should go into a separate method and not into `ConfigureRehearsalJobs`. Let's keep CRJ's concern to turning jobs into rehearsals, it's already quite big. Let the new method return `config.Presubmits` and merge them at the call site. The new method would be also more testable.

Reply:
addressed

Reply:
What I actually had in mind (but haven't described well, re-reading myself) was even more extreme - basically entirely remove this path from `ConfigureRehearsalJobs`. Call it from `main()`. That way `CRJ` can be kept simpler - no need to add new param to it, less responsibilities overall...

But I guess this is fine too.

Reply:
Its need to be here in order to exclude if any changed job is using the specific template with the specific cluster_type. I agree that it would be better to have this outside, but it will need more refactoring. I could open a follow-up for it.

Reply:
Unless I'm overlooking something, the block [here](https://github.com/openshift/ci-operator-prowgen/pull/122/files#diff-fe857c8e1c6720bb366e42d076a9b101R228) simply adds more rehearsals, where's the exclusion happening?

Reply:
oh, Jesus, I removed it when splitting the logic. Nice catch. I will also split the logic to `main` as you suggested.

Reply:
addressed